provider/azure/cache.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"time"
+)
+
+// zonesCache is a cache for Azure zones(private or public)
+type zonesCache[T any] struct {
+	age      time.Time
```

Comment:
This we don't need, you can add a `now func() time.Time` to use your own time in tests.

Reply:
This is not just for test but part of the logic as well. Please correct me if I am missing something.